core/remote/change.js

```diff
@@ -121,6 +124,11 @@ const isTrash = (a /*: RemoteChange */) /*: boolean %checks */ => a.type === 'Di
 const isRestore = (a /*: RemoteChange */) /*: boolean %checks */ => a.type === 'DirRestoration' || a.type === 'FileRestoration'
 const isDissociate = (a /*: RemoteChange */) /*: boolean %checks */ => a.type === 'DirDissociation' || a.type === 'FileDissociation'
 
+function includeDescendant (parent /*: RemoteDirMove */, e /*: RemoteDescendantChange */) {
+  parent.descendantMoves = parent.descendantMoves || []
+  parent.descendantMoves.push(e, ...(e.descendantMoves || []))
```

Comment:
Wouldn't concat be more efficient?
Maybe `delete e.descendantMoves` afterwards to prevent confusion when looking at the data?

Reply:
`concat` is less efficient (constructing new arrays) and IMHO, less readable : 

`parent.descendantMoves = (parent.descendantMoves || []).concat(e).concat(e.descendantMoves || [])`

deleting further descendantMoves is loosing debug data, better to have them and not need them.

Reply:
> deleting further descendantMoves is loosing debug data, better to have them and not need them.

Don't we already have `ancestorPath` for this?
Or do we believe it won't help debugging?

Reply:
Added `delete e.descendantMoves`